### PR TITLE
In CTE Null first order behavior is not correct

### PR DESCRIPTION
**Pull request by forestkeeper**
Previously we add the Null first check after analyze, and in that case it'll miss to add the Null first tag to the sort clause for a CTE query

change the place adding Null first tag to the place sortClause is initialized

Task: BABEL-3991
Signed-off-by: Zhibai Song <szh@amazon.com>

### Description

the following sql not produce the right order for NULL : 
```
drop table if exists t1;
create table t1(a int);
insert t1 values (1);
insert t1 values (3);
insert t1 values (null);
with t1cte AS (
select top(3) a from t1 order by 1
)
select * from t1cte; -- Returns NULL at bottom not expected
select top(3) a from t1 order by 1; -- Returns NULL at the TOP as expected
```

the root cause is we deal with sort clause inproperly

### Issues Resolved

Will return the right order for null include 
- asc order by return NULL first
- no order by , natural asc, return NULL last
- desc order by return NULL last

checked this behavior with 
- simple select
- select order by
- subquery order by 
- cte with inner query order by
- babel_functions
- etc other remainning explain query

### Check List
- [ x ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/JDBC/expected/babel_function.out [hunk 1]
```diff
@@ -2426,26 +2426,26 @@ SELECT a, b, COUNT(*) OVER (ORDER BY a) from t3;
 GO
```
Comment (forestkeeper) [inline test/JDBC/expected/babel_function.out:2426]: order by a , nullfirst +  asc , null comes first

File: test/JDBC/expected/babel_function.out [hunk 2]
```diff
@@ -2426,26 +2426,26 @@ SELECT a, b, COUNT(*) OVER (ORDER BY a) from t3;
...
+efg#!#b#!#5
+xyz#!#a#!#7
+xyz#!#b#!#7
 ~~END~~
 
 SELECT a, b, COUNT(*) OVER (ORDER BY a DESC) from t3;
 GO
 ~~START~~
 varchar#!#varchar#!#int
-<NULL>#!#<NULL>#!#1
```
Comment (forestkeeper) [inline test/JDBC/expected/babel_function.out:2442]: order by desc , null comes last

File: contrib/babelfishpg_tsql/src/hooks.c [hunk 3]
```diff
@@ -3630,3 +3636,13 @@ bbf_check_rowcount_hook(int es_processed)
 	else
 		return false;
 }
+
+static void 
+sort_nulls_first(SortGroupClause * sortcl, bool reverse)
+{
+	if (sql_dialect == SQL_DIALECT_TSQL)
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/hooks.c:3643]: To test this dialect check, could you also add a test case in PG connection?

Comment (forestkeeper) [reply contrib/babelfishpg_tsql/src/hooks.c:3643]: OK

**Review by 2jungkook — APPROVED** [review]

**Pull request merged**